src/cascade/input_data/db/configuration.py

```diff
@@ -2,30 +2,111 @@
 
-def settings_for_model(meid=None, mvid=None):
-    if len([c for c in [meid, mvid] if c is not None]) != 1:
-        raise ValueError("Must supply either mvid or meid")
 
-    if mvid is not None:
-        mvid = mvid
-    else:
-        modelable_entity_id = meid
-        ec = make_execution_context(modelable_entity_id=modelable_entity_id)
-        mvid = latest_model_version(ec)
-        CODELOG.info(f"No model version specified so using the latest version for model {meid} which is {mvid}")
+def load_raw_settings_meid(modelable_entity_id):
+    """
+    Given a meid, get settings for the latest corresponding mvid.
+
+    Args:
+        modelable_entity_id (int,str): The MEID.
+    Returns:
+        dict: Settings as a JSON dictionary of dictionaries.
+        int: Model version ID that is the latest one associated with this meid.
+    """
+    ec = make_execution_context(modelable_entity_id=modelable_entity_id)
+    mvid = latest_model_version(ec)
+    MATHLOG.info(
+        f"No model version specified so using the latest version for "
+        f"model {modelable_entity_id} which is {mvid}")
+    ec = make_execution_context(model_version_id=mvid)
+    raw_settings = settings_json_from_epiviz(ec)
+    return raw_settings, mvid
+
+
+def load_raw_settings_mvid(mvid):
+    """Given an mvid, get its settings.
+
+    Args:
+        mvid (int,str): Model version ID.
+    Returns:
+        dict: Settings as a JSON dictionary of dictionaries.
+        int: Model version ID that was passed in.
+    """
     ec = make_execution_context(model_version_id=mvid)
     raw_settings = settings_json_from_epiviz(ec)
+    return raw_settings, mvid
+
+
+def load_raw_settings_file(settings_file):
+    """Given a settings file, get the latest mvid for its meid.
 
-    if "model_version_id" not in raw_settings["model"] or not raw_settings["model"]["model_version_id"]:
+    Args:
+        settings_file (str): Model version ID.
+    Returns:
+        dict: Settings as a JSON dictionary of dictionaries.
+        int: Model version ID, either found in file or latest from meid.
+    """
+    with open(str(settings_file), "r") as f:
+        raw_settings = json.load(f)
+    if "model" in raw_settings and "modelable_entity_id" in raw_settings["model"]:
+        ec = make_execution_context(modelable_entity_id=raw_settings["model"]["modelable_entity_id"])
+    else:
+        raise SettingsError(
+            f"The settings file should have a modelable_entity_id in it. "
+            f"It would be under model and then modelable_entity_id.")
+    if "model_version_id" in raw_settings["model"]:
+        mvid = raw_settings["model"]["model_version_id"]
+        MATHLOG.info(f"Using mvid {mvid} from the settings file.")
+    else:
+        mvid = latest_model_version(ec)
+        MATHLOG.info(f"Using mvid {mvid} from latest model version.")
+    return raw_settings, mvid
+
+
+def json_settings_to_frozen_settings(raw_settings, mvid=None):
```

Comment:
Yes, good to extract this from the executor code.

Reply:
I thought the executor was the thing that ran Dismod-AT. Now it's any kind of main() plus the thing that ran Dismod-AT. I want to reorganize.